control_panel_api/models.py

```diff
@@ -234,24 +234,37 @@ class Meta:
     def aws_role_name(self):
         return self.app.iam_role_name
 
+    def has_readwrite_access(self):
+        return self.access_level == self.READWRITE
+
 
 class UserS3Bucket(AccessToS3Bucket):
     """
     A user can have access to several S3 buckets.
 
     We have two access levels, "readonly" (default) and "readwrite".
-    The `is_admin` field determine if the user has admin privileges on the
-    S3 bucket
-    """
 
+    The `is_admin` field determines the `access_level` to the S3Bucket. The
+    `access_level` can not be changed directly.
+    """
     user = models.ForeignKey(
         User, related_name='users3buckets', on_delete=models.CASCADE)
     is_admin = models.BooleanField(default=False)
 
     class Meta:
-        # one record per user/s3bucket
         unique_together = ('user', 's3bucket')
         ordering = ('id',)
 
     def aws_role_name(self):
         return self.user.iam_role_name
+
+    def has_readwrite_access(self):
```

Comment:
I don't think this is a good name for what the method is doing. This is checking if the user can admin the bucket, not if he can read/write on it.

I know it seems like an admin of an S3 bucket can also read and write but not necessarily.
I may be an admin for an S3 bucket containing sensitive information but not having the ability to read and write it.

Reply:
Suggestions?  As I'm just adapting existing code and template method.

Reply:
I think maybe this method doesn't need to be overridden

Reply:
I see what you mean about the template method pattern. I still think the access to S3 bucket would be determined by the `access_level` field - I don't remember adding this `had_readwrite_access` method but doesn't really matter - I imagine implementation is the same for both `AppS3Bucket` and `UserS3Bucket`:

```python
    def has_readwrite_access(self):
        return self.access_level == self.READWRITE
```